### PR TITLE
Bump mypy from 1.10.1 to 1.11.0

### PR DESCRIPTION
**Pull request by JuliaBars**
# I have made things!

## Checklist

- [ ] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ ] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [ ] I have added my changes to the `CHANGELOG.md`

## Related issues

🙏 Please, if you or your company finds `dry-python` valuable, help us sustain the project by sponsoring it transparently on https://github.com/sponsors/dry-python. As a thank you, your profile/company logo will be added to our main README which receives hundreds of unique visitors per day.

### CODE REVIEW

File: typesafety/test_examples/test_your_container/test_pair4_def.yml [hunk 1]
```diff
@@ -23,5 +23,5 @@
     my_pair: Pair[int, str] = Pair.from_paired(1, 'a')
     reveal_type(my_pair.pair(function))
   out: |
-    main:4: note: Revealed type is "def [_FirstType, _SecondType] (inner_value: Tuple[_FirstType`1, _SecondType`2]) -> test_pair4.Pair[_FirstType`1, _SecondType`2]"
-    main:10: note: Revealed type is "test_pair4.Pair[builtins.float, builtins.bool]"
+    main:4: note: Revealed type is "Any"
```
Comment (sobolevn) [inline typesafety/test_examples/test_your_container/test_pair4_def.yml:26]: You should not change expected types to `Any`, almost never. This basically means that there's a type error somewhere. Please, revert this change.

File: typesafety/test_examples/test_your_container/test_pair4_reuse.yml [hunk 2]
```diff
@@ -19,5 +19,5 @@
     reveal_type(my_pair.map(str))
     reveal_type(map_(str)(my_pair))
   out: |
-    main:5: note: Revealed type is "test_pair4.Pair[builtins.str, builtins.int]"
-    main:6: note: Revealed type is "test_pair4.Pair[builtins.str, builtins.int]"
+    main:5: note: Revealed type is "Any"
```
Comment (sobolevn) [inline typesafety/test_examples/test_your_container/test_pair4_reuse.yml:22]: The same here.

File: pyproject.toml [hunk 3]
```diff
@@ -49,7 +49,7 @@ _ = "returns.contrib.hypothesis._entrypoint"
 python = "^3.9"
 
 typing-extensions = ">=4.0,<5.0"
-mypy = { version = ">=1.10,<1.11", optional = true }
+mypy = "1.11.0"
```
Comment (sobolevn) [inline pyproject.toml:52]:
```suggestion
mypy = mypy = { version = ">=1.11,<1.12", optional = true }
```

Comment (sobolevn) [reply pyproject.toml:52]: You would also need to run `poetry update` again

File: typesafety/test_maybe/test_maybe_type.yml [hunk 4]
```diff
@@ -142,9 +142,9 @@
   disable_cache: false
   main: |
     from returns.maybe import Maybe
-    from typing import NoReturn
```
Comment (sobolevn) [inline typesafety/test_maybe/test_maybe_type.yml:145]: This change is dangerous, because not python versions have `typing` with `Never`, please use `typing_extensions`.

**Review by sobolevn — COMMENTED** [review]
Thanks a lot! Almost done!

File: typesafety/test_examples/test_your_container/test_pair4_def.yml (hunk 1 above)
Comment (JuliaBars) [reply typesafety/test_examples/test_your_container/test_pair4_def.yml]: done

File: typesafety/test_examples/test_your_container/test_pair4_reuse.yml (hunk 2 above)
Comment (JuliaBars) [reply typesafety/test_examples/test_your_container/test_pair4_reuse.yml]: done

File: pyproject.toml (hunk 3 above)
Comment (JuliaBars) [reply pyproject.toml]: done

File: typesafety/test_maybe/test_maybe_type.yml (hunk 4 above)
Comment (JuliaBars) [reply typesafety/test_maybe/test_maybe_type.yml:145]: done

File: typesafety/test_examples/test_your_container/test_pair4_error.yml [hunk 5]
```diff
@@ -21,5 +21,3 @@
 
     my_pair = Pair.from_paired(1, 'a')
     my_pair.pair(function)  # this should and will error
-  out: |
```
Comment (sobolevn) [inline typesafety/test_examples/test_your_container/test_pair4_error.yml:24]: This should not be removed.

File: typesafety/test_examples/test_your_container/test_pair4_error.yml [hunk 6]
```diff
@@ -21,5 +21,3 @@
 
     my_pair = Pair.from_paired(1, 'a')
     my_pair.pair(function)  # this should and will error
```
Comment (sobolevn) [inline typesafety/test_examples/test_your_container/test_pair4_error.yml:23]:
```suggestion
    my_pair.pair(function)  # this should and will error
  out: |
    main:9: error: Argument 1 to "pair" of "Pair" has incompatible type "Callable[[str, int], Pair[float, bool]]"; expected "Callable[[int, str], KindN[Pair[Any, Any], float, bool, Any]]" [arg-type]
```

**Review by sobolevn — APPROVED** [review]
Thanks a lot! Congrats on your first `dry-python` PR 🎉
